src/frontend/src/storage/index.test.ts

```diff
@@ -117,23 +115,46 @@ test(
 );
 
 test(
-  "anchors are also written to localstorage",
+  "V2 anchors are migrated",
```

Comment:
I feel like for the complexity and size of this PR, the storage V3 is really light on tests.
I.e. there are no test for:
- principal storage / lookup
- making sure many principals are stored
- making sure the principals are actually hashed

Reply:
Yeah, that's not incorrect. Some of this is tested by the e2e tests but I'm happy to add some tests.